datanode/entities/aggregated_balance.go

```diff
@@ -138,6 +138,7 @@ func (balance *AggregatedBalance) ToProto() *v2.AggregatedBalance {
 }
 
 func (balance AggregatedBalance) Cursor() *Cursor {
+	// though this should be OK, it's not exactly safe for concurrent use
```

Comment:
What do you mean?

Reply:
That comment was more for my own sanity while going through the entities package. Technically, because the ID fields in the `AggregateBalance` type are pointers, and creating an `AggregatedBalanceCursor`, like we are doing here, is not an atomic operation, so _in theory_ this could be a problem if we were to alter the values we point to concurrently to calling this `Cursor` function. In practice, however, this is a non-issue. Just left the comment there to indicate (to myself) that I'd already gone through this file

Reply:
Maybe I'm missing a something here - but I don't we what using a pointer has to do with it; I agree that there's no synchronisation here so it's not safe to make a cursor while you're altering the entity itself; but that that's the case with all the entities and indeed, most go structs?

If I were wanting to share these between threads (which I'm pretty sure we don't) I'd probably expect the caller to manage the synchronisation.

I guess my question is what would 'OK' look like for you here? If it's not OK in some way I don't understand we should fix it, otherwise I think we should remove the comment because as-is when reading it it sows Fear, Uncertainty and Doubt in the heart of the reader ;)

Reply:
I don't think this is resolved.